File: java/src/main/java/it/giannibombelli/wsc2026/giftcard/domain/giftcard/GiftCardId.java
package it.giannibombelli.wsc2026.giftcard.domain.giftcard;

import it.giannibombelli.wsc2026.common.utils.Require;

import it.giannibombelli.wsc2026.common.domain.identity.AggregateId;

import java.util.UUID;

public record GiftCardId(UUID value) implements AggregateId {
    public GiftCardId {
        Require.requireArgument(value, "GiftCardId value");
    }
}

File: typescript/src/giftcard/domain/giftcard/giftCardId.ts
// Identità dell'aggregato GiftCard.

import { requireArgument } from '@/common/utils/requireArgument.js';
import type { AggregateId } from '@/common/domain/identity/aggregateId.js';
import { Uuid } from '@/common/domain/primitive/uuid.js';

export class GiftCardId implements AggregateId {
  readonly value: Uuid;

  constructor(value: Uuid) {
    requireArgument(value, 'GiftCardId value');
    this.value = value;
  }
}
